src/analyze.cpp

```diff
@@ -3069,30 +3068,50 @@ static Error resolve_union_zero_bits(CodeGen *g, ZigType *union_type) {
 
     union_type->data.unionation.resolve_loop_flag_zero_bits = true;
 
-    assert(union_type->data.unionation.fields == nullptr);
-    uint32_t field_count = (uint32_t)decl_node->data.container_decl.fields.length;
-    if (field_count == 0) {
-        add_node_error(g, decl_node, buf_sprintf("unions must have 1 or more fields"));
+    uint32_t field_count;
+    if (decl_node->type == NodeTypeContainerDecl) {
+        assert(union_type->data.unionation.fields == nullptr);
+        field_count = (uint32_t)decl_node->data.container_decl.fields.length;
+        if (field_count == 0) {
+            add_node_error(g, decl_node, buf_sprintf("unions must have 1 or more fields"));
```

Comment:
This should also be checked for generated unions. Seems to also be missing from enums.